mods/gen4/random-teams.js

```diff
@@ -61,6 +61,14 @@ class RandomGen4Teams extends RandomGen5Teams {
 			'Adaptability':1, 'Hustle':1, 'Iron Fist':1, 'Skill Link':1,
 		};
 
+		// Give recovery moves priority over certain other defensive status moves
+		let recoveryMoves = {
+			'healorder':1, 'milkdrink':1, 'moonlight':1, 'morningsun':1, 'recover':1, 'rest':1, 'roost':1, 'slackoff':1, 'softboiled':1, 'synthesis':1, 'wish':1,
```

Comment:
Hmm, why not? I included it for things like Suicune and Snorlax because 3 attacks + Roar/WW is a pretty bad set on them (not terrible, but definitely worse than most other sets they could get). They really only want Roar/WW with RestTalk.

There's also a couple of Pokemon that have both RestTalk and regular recovery in their movepools (Milotic, Slowbro, Cradily) and if we don't include Rest here then a set like Surf / Toxic / Rest / Sleep Talk would get rejected because the check wouldn't count Rest as a recovery move and would see that Recover is available.